include/irr/video/decodePixels.h

```diff
@@ -1293,6 +1307,16 @@ namespace irr { namespace video
             for (uint32_t i = 0; i < 3u; ++i)
                 lin[i] = s[i] * 255.;
         }
+
+        template<typename T>
```

Comment:
templating is OK, but you need a `static_assert` which ensures that `T` can only be `float` or `double`

Reply:
@Florastamine, we need a static_assert, dont want someone using that with ints or vector classes